### ISSUE TITLE
[Bug]Fps Hidding Objects/Renderers

### ISSUE THREAD

**Issue by moepi2k** [suriyun-production/mmorpg-kit-docs#1667]
When use shooter controller with FPS, when login the  Hidding Objects/Renderes for FPS is not triggered
https://gyazo.com/2d8e3066a45061f529b1a120fc2050ba

it only triggers when switch from tps to fps

p.S this happen only in singleplayer. in mmo it works

**Comment by insthync** [comment]
As I try, it still works properly.

You may try debugging `CharacterModelManager` -> `SetIsFps`, `UpdateVisibleState`

**Comment by insthync** [comment]
Ahhh, I see, the weird thing is it is set to be hidden but it is not, seems like it is Unity bug.

**Comment by insthync** [comment]
Try replace `SetHiddingObjectsAndRenderers` function with 

```
        protected async void SetHiddingObjectsAndRenderers(GameObject[] hiddingObjects, Renderer[] hiddingRenderers, bool isHidding)
        {
            await Cysharp.Threading.Tasks.UniTask.NextFrame();
            int i;
            if (hiddingObjects != null && hiddingObjects.Length > 0)
            {
                for (i = 0; i < hiddingObjects.Length; ++i)
                {
                    hiddingObjects[i].SetActive(!isHidding);
                }
            }
            if (hiddingRenderers != null && hiddingRenderers.Length > 0)
            {
                for (i = 0; i < hiddingRenderers.Length; ++i)
                {
                    hiddingRenderers[i].forceRenderingOff = isHidding;
                }
            }
        }
```

**Comment by insthync** [comment]
Fixed in 1.77
